### DISCUSSION TITLE
Converting key-values pairs

### DISCUSSION THREAD

**Discussion by KrishnaJyothika** [vectordotdev/vector#19655] [General] [answered]
Hi Team,

We have an requirement where we want to convert one of the JSON field into key-value pairs

Here we want to convert value of **resource** into key-value pairs

**Input**:
{"Name":"List","Tier":"Hot","average":20,"count":1,"maximum":1,"metricName":"Availability","minimum":100,"resource":"**/key1/value1/key2/value2/key3/values3**","time":"2023-12-14T13:22:00.0000000Z","total":1}

**Excepted Output**:
{"Name":"List","Tier":"Hot","average":20,"count":1,"maximum":1,"metricName":"Availability","minimum":100,**"key1":"value1","key2":"value2","key3":"value3"**,"time":"2023-12-14T13:22:00.0000000Z","total":1}

Please help us understand how can we achieve this using vector.

**Comment by jszwedko** [comment]
Unfortunately I don't think that sort of thing would be possible in VRL. You should be able to use a Lua transform though.

**Comment by jlambatl** [comment, marked as answer]
@KrishnaJyothika here's a way to this.. 

Here's a playground [link](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoia2V5VmFsUmVnZXggPSByJy8oP1A8a2V5PlteL10rKS8oP1A8dmFsdWU%2BW14vXSspJ1xub2JqID0gb2JqZWN0KC4pXG5cbnBhaXJzLCBlcnIgPSBwYXJzZV9yZWdleF9hbGwob2JqLnJlc291cmNlLCBrZXlWYWxSZWdleClcbmlmIGVyciAhPSBudWxsIHtcbiAgICBhYm9ydFxufVxuZm9yX2VhY2gocGFpcnMpIC0%2BIHwgX2lkeCwgbWF0Y2ggfCB7XG4gICAgayA9IHRvX3N0cmluZyhtYXRjaC5rZXkpXG4gICAgdiA9IHRvX3N0cmluZyhtYXRjaC52YWx1ZSlcbiAgICBvYmosIGVyciA9IHNldCh2YWx1ZTogb2JqLCBwYXRoOiBbIGsgXSwgZGF0YTogdilcbiAgICBpZiBlcnIgIT0gbnVsbCB7XG4gICAgICAgIGFib3J0XG4gICAgfVxufVxuZGVsKG9iai5yZXNvdXJjZSlcblxuLiA9IG9iaiIsImV2ZW50Ijp7Ik5hbWUiOiJMaXN0IiwiVGllciI6IkhvdCIsImF2ZXJhZ2UiOjIwLCJjb3VudCI6MSwibWF4aW11bSI6MSwibWV0cmljTmFtZSI6IkF2YWlsYWJpbGl0eSIsIm1pbmltdW0iOjEwMCwicmVzb3VyY2UiOiIva2V5MS92YWx1ZTEva2V5Mi92YWx1ZTIva2V5My92YWx1ZXMzIiwidGltZSI6IjIwMjMtMTItMTRUMTM6MjI6MDAuMDAwMDAwMFoiLCJ0b3RhbCI6MX0sImlzX2pzb25sIjpmYWxzZSwiZXJyb3IiOm51bGx9).

Here's the example, somewhat verbose but it does work.

```
keyValRegex = r'/(?P<key>[^/]+)/(?P<value>[^/]+)'
obj = object(.)

pairs, err = parse_regex_all(obj.resource, keyValRegex)
if err != null {
    abort
}
for_each(pairs) -> | _idx, match | {
    k = to_string(match.key)
    v = to_string(match.value)
    obj, err = set(value: obj, path: [ k ], data: v)
    if err != null {
        abort
    }
}
del(obj.resource)

. = obj
```
